integration-test/mqtt/mqtt_system_test.c

```diff
@@ -1170,20 +1168,17 @@ void test_MQTT_Restore_Session_Complete_Incoming_Publish( void )
 
     /* Disconnect on receiving PUBREL so that we are not able to complete in the incoming QoS2
      * PUBLISH in the current connection. */
-    disconnectOnPacketType = MQTT_PACKET_TYPE_PUBREL;
+    packetTypeForDisconnection = MQTT_PACKET_TYPE_PUBREL;
     TEST_ASSERT_EQUAL( MQTTSendFailed,
                        MQTT_ProcessLoop( &context, 3 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
 
-    /* Verify that the connection with the broker has been disconnected. */
-    TEST_ASSERT_EQUAL( MQTTNotConnected, context.connectStatus );
```

Comment:
?

Reply:
Same as [this](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1252#discussion_r489106292)